docs/recipes/custom-useatom-hooks.mdx

```diff
@@ -14,16 +14,25 @@ This page shows the ways of creating different utility functions. Utility functi
 import { useAtomValue } from 'jotai'
 import { selectAtom } from 'jotai/utils'
 
-/* if an atom is created here, please use `useMemo(() => atom(initValue), [initValue])` instead. */
-export function useSelectAtom(anAtom, keyFn) {
-  return useAtomValue(selectAtom(anAtom, keyFn))
+export function useSelectAtom(anAtom, selector, deps) {
+  const selectorAtom = useMemo(
+    () =>
+      selectAtom(
+        anAtom,
+        (v, prevSlice) => selector(v, prevSlice),
+        // alternatively, you can customize `equalityFn` to determine when it will rerender
+        undefined,
+      ),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    deps,
```

Comment:
We don't recommend passing through `deps`, as eslint rules can't catch it.
Please use `useCallback` at the caller end.